contracts/utils/AddressUtils.sol

```diff
@@ -10,6 +10,14 @@ library AddressUtils {
     error AddressUtils__InsufficientBalance();
     error AddressUtils__NotContract();
     error AddressUtils__SendValueFailed();
+    error AddressUtils__FailedCall();
+    error AddressUtils__FailedCallWithValue();
+
+    // TODO: without this function, custom errors referenced by selector are not available in tests
+    function __placeholder() internal {
+        revert AddressUtils__FailedCall();
+        revert AddressUtils__FailedCallWithValue();
+    }
```

Comment:
This may be a problem with `hardhat-chai-matchers`, as the errors appear in the `AddressUtils` ABI even without this function.

Reply:
Actually, they don't appear in `$AddressUtils`, which is the contract used in the tests.  They're not inherited because `AddressUtils` is a `library`.  I think this is a Solidity issue, though arguably it's expected behavior.